scipy/stats/_fisher_exact.py

```diff
@@ -0,0 +1,505 @@
+    """
+    Conditional exact confidence interval for the odds ratio.
+
+    This function implements Cornfield's "exact confidence limits",
+    as explained in section 2 of the paper:
+
+        J. Cornfield (1956), A statistical problem arising from
+        retrospective studies. In Neyman, J. (ed.), Proceedings of
+        the Third Berkeley Symposium on Mathematical Statistics and
+        Probability 4, pp. 135-148.
+
+    """
+    if alternative == 'two-sided':
+        alpha = 0.5*(1 - confidence_level)
+        lower = _ci_lower(table, alpha)
+        upper = _ci_upper(table, alpha)
+    elif alternative == 'less':
+        lower = 0.0
+        upper = _ci_upper(table, 1 - confidence_level)
+    else:
+        # alternative == 'greater'
+        lower = _ci_lower(table, 1 - confidence_level)
+        upper = np.inf
+
+    return lower, upper
+
+
+FisherExactResult = _make_tuple_bunch('FisherExactResult',
+                                      ['sample_odds_ratio', 'pvalue'],
+                                      ['conditional_odds_ratio',
+                                       'conditional_odds_ratio_ci'])
+ConfidenceInterval = namedtuple('ConfidenceInterval', ['low', 'high'])
+
+
+def fisher_exact(table, alternative='two-sided', confidence_level=0.95):
+    """
+    Perform a Fisher exact test on a 2x2 contingency table.
+
+    Parameters
+    ----------
+    table : array_like of ints
+        A 2x2 contingency table.  Elements should be non-negative integers.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available (default is 'two-sided'):
+
+          * 'two-sided'
+          * 'less': one-sided
+          * 'greater': one-sided
+
+    confidence_level: float
+        Desired confidence level for the confidence interval that is stored
+        as the ``conditional_odds_ratio_ci`` attribute of the return value.
+        The value must be given as a fraction between 0 and 1.
+        Default is 0.95 (meaning 95%).
+
+    Returns
+    -------
+    result : object
+        The returned object has four attributes:
+
+          * ``sample_odds_ratio``, float, is
+            ``table[0, 0]*table[1, 1]/(table[0, 1]*table[1, 0])``.
+            This is the prior odds ratio and not a posterior estimate.
+          * ``pvalue``, float, is the probability of obtaining a
+            distribution at least as extreme as the one that was
+            actually observed, assuming that the null hypothesis is
+            true.
+          * ``conditional_odds_ratio``, float, is the conditional
+            maximum likelihood estimate for the odds ratio.   It is
+            the noncentrality parameter of Fisher's noncentral
+            hypergeometric distribution with the same hypergeometric
+            parameters as ``table`` whose mean is ``table[0, 0]``.
+            (FIXME: I know this need work!)
```

Comment:
Does it still?

Reply:
It could use a little copy-editing.  I'll tweak the docstring in my next update.